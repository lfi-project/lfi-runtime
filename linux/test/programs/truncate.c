#include <stdio.h>
#include <unistd.h>
#include <fcntl.h>
#include <string.h>

int main(int argc, char **argv) {
    (void) argc;
    const char *filename = argv[1];
    const char *original_content = "This is the original content of the file.\n";
    size_t original_len = strlen(original_content);

    // Step 1: Write original content to the file
    int fd = open(filename, O_CREAT | O_WRONLY | O_TRUNC, 0644);
    if (fd == -1) {
        perror("open");
        return 1;
    }
    write(fd, original_content, original_len);
    close(fd);
    printf("Original content written to '%s'.\n", filename);

    // Step 2: Truncate the file to a shorter length
    if (truncate(filename, 10) == 0) {
        printf("File '%s' truncated to 10 bytes.\n", filename);
    } else {
        perror("truncate");
        return 1;
    }

    // Step 3: Restore the original content
    fd = open(filename, O_WRONLY | O_TRUNC);
    if (fd == -1) {
        perror("open (restore)");
        return 1;
    }
    write(fd, original_content, original_len);
    close(fd);
    printf("Original content restored in '%s'.\n", filename);

    return 0;
}

