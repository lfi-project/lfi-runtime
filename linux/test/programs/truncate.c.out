Original content written to 'programs/inputs/truncate.txt'.
File 'programs/inputs/truncate.txt' truncated to 10 bytes.
Original content restored in 'programs/inputs/truncate.txt'.
